File: XN/pset2/readability.c
#include <stdio.h>
#include <ctype.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main()
{
    string text;
    int letter = 0, word = 1, sentence = 0;
    int i;
    float index;

    text = get_string("Text: ");
// traverse the string to identify letters, words, and sentences via various features stated in the problem
    for (i = 0; i < strlen(text); i++){
        if(isalpha(text[i])){
            letter ++;
        }
        else if (text[i] == ' '){
            word ++;
        }
        else if (text[i] == '.'||text[i] == '?'||text[i] == '!'){
            sentence ++;
        }
    }
    /*printf("%d letter(s)\n", letter);
    printf("%d word(s)\n", word);
    printf("%d sentence(s)\n", sentence);
    */
    
    index = 0.0588 * ((float)letter/(float)word) * 100 - 0.296 * ((float)sentence/(float)word) * 100 - 15.8;
    //printf ("%f \n", index);
    if(index >= 1 && index < 16){
        printf("Grade %d\n", (int)round(index));
    }
    else if (index >= 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Before Grade 1\n");
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

text = None

## Repeatedly promt user for input
while (text == None):
    text = get_string("Text:  ")

## Initialize letter, word and sentence count
letter = 0
word = 1
sentence = 0

## Counte by features
for char in text:
    if char.isalpha():
        letter += 1
    
    elif char == ' ':
        word += 1
    
    elif char == '.' or char == '?' or char == '!':
        sentence += 1

## Calculate by formula
index = 0.0588 * (letter/word) * 100 - 0.296 * (sentence/word) * 100 - 15.8

## Round index to grade
grade = round(index)

## Print output
if grade > 0 and grade < 17:
    print("Grade", grade)
elif grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")